src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3081,7 +3081,7 @@ private static ExpressionSyntax ParenthesizeLeft(ExpressionSyntax expression)
                 return expression;
             }
 
-            return Parenthesize(expression);
+            return (ExpressionSyntax)Parenthesize(expression);
```

Comment:
helper method can now take in expressions/patterns and return expressions/patterns.